### PR TITLE
Feat(eos_cli_config_gen): Add WRED support

### PR DESCRIPTION
**Pull request by chetryan**
## Change Summary

At the moment for Weighted Random Early Detection to be enabled on certain queues, we would need to raw cli commands or custom templates.

This change adds support to define WRED on a per queue bases using the AVD schema.

## Related Issue(s)

- No reported issues

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

This change introduces:
- a schema for defining WRED per queue in qos-profiles
- a schema for defining WRED per queue on ethernet interfaces
- associated jinja templates to generate documentation and eos config

## How to test

`molecule test -s eos_cli_config_gen`

### User Checklist

- Review updated documentation (using `make webdoc-up` to locally host the updated documentation
- Review the molecule test case and expected output

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Thank you for another great PR.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/qos.cfg [hunk 1]
```diff
@@ -112,6 +112,60 @@ qos profile uc_mc_queues_test
...
+   tx-queue 1
+      !! Test no priority
+      bandwidth percent 50
+      no priority
+      random-detect drop minimum-threshold 1 kbytes maximum-threshold 10 kbytes drop-probability 100
+   !
+   tx-queue 2
+      bandwidth percent 10
+      priority strict
+      random-detect drop drop-precedence 2 minimum-threshold 2 kbytes maximum-threshold 200 kbytes
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/qos.cfg]:
```
ceos2(config-qos-profile-wred_queues_test)#   tx-queue 1
ceos2(config-qos-profile-wred_queues_test-txq-1)#      !! Test no priority
ceos2(config-qos-profile-wred_queues_test-txq-1)#      bandwidth percent 50
ceos2(config-qos-profile-wred_queues_test-txq-1)#      no priority
ceos2(config-qos-profile-wred_queues_test-txq-1)#      random-detect drop minimum-threshold 1 kbytes maximum-threshold 10 kbytes drop-probability 100
ceos2(config-qos-profile-wred_queues_test-txq-1)#   !
ceos2(config-qos-profile-wred_queues_test-txq-1)#   tx-queue 2
ceos2(config-qos-profile-wred_queues_test-txq-2)#      bandwidth percent 10
ceos2(config-qos-profile-wred_queues_test-txq-2)#      priority strict
ceos2(config-qos-profile-wred_queues_test-txq-2)#      random-detect drop drop-precedence 2 minimum-threshold 2 kbytes maximum-threshold 200 kbytes
% Incomplete command
```

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/qos.cfg]:
```
ceos2(config-qos-profile-wred_queues_test-mc-txq-2)# random-detect drop drop-precedence 2 minimum-threshold 2 kbytes maximum-threshold 200 ?
  kbytes  Set threshold in kbytes

ceos2(config-qos-profile-wred_queues_test-mc-txq-2)# random-detect drop drop-precedence 2 minimum-threshold 2 kbytes maximum-threshold 200 kbytes ?
  drop-probability  Set maximum drop probability rate

ceos2(config-qos-profile-wred_queues_test-mc-txq-2)# random-detect drop drop-precedence 2 minimum-threshold 2 kbytes maximum-threshold 200 kbytes drop-probability ?
  <1-100>  Specify maximum drop rate

ceos2(config-qos-profile-wred_queues_test-mc-txq-2)# random-detect drop drop-precedence 2 minimum-threshold 2 kbytes maximum-threshold 200 kbytes drop-probability 42 ?
  weight  Set WRED weight
  <cr>

```

**Review by gmuloc — CHANGES_REQUESTED** [review]
Looks like stoping after the `maximum-theshold` without a `drop-probablity` is not allowed

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos_profiles.schema.yml [hunk 2]
```diff
@@ -128,6 +128,46 @@ keys:
...
+                              - str
+                            description: WRED minimum-threshold
+                          max:
+                            type: int
+                            required: true
+                            min: 1
+                            convert_types:
+                              - str
+                            description: WRED maximum-threshold
+                          drop_probability:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos_profiles.schema.yml:163]: should this be made required then?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos_profiles.schema.yml [hunk 3]
```diff
@@ -128,6 +128,46 @@ keys:
...
+                            type: int
+                            min: 1
+                            max: 100
+                            convert_types:
+                              - str
+                            description: WRED drop probability.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos_profiles.schema.yml:170]: probably not your use case here but why not add `weight <0-15>` as well?

```
ceos2(config-qos-profile-wred_queues_test-txq-2)#random-detect drop drop-precedence 1 minimum-threshold 23 bytes max 23 bytes drop-probability 1 ?
  weight  Set WRED weight
  <cr>

ceos2(config-qos-profile-wred_queues_test-txq-2)#random-detect drop drop-precedence 1 minimum-threshold 23 bytes max 23 bytes drop-probability 1 weight ?
  <0-15>  Value of WRED weight

```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/qos.cfg (hunk 1 above)
Comment (chetryan) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/qos.cfg]: Agreed that this is an issue. Updated the schema

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos_profiles.schema.yml (hunk 2 above)
Comment (chetryan) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos_profiles.schema.yml:163]: Agreed, updated.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos_profiles.schema.yml (hunk 3 above)
Comment (chetryan) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos_profiles.schema.yml:170]: Added this now to both schema and templates

**Review by carlbuchmann — APPROVED** [review]
LGTM

- tested congis on dut
- review schema + documentation
- all comments from Claus and Guillaume addressed

**Pull request merged**